cmd/influx/config.go

```diff
@@ -60,12 +60,21 @@ func (b *cmdConfigBuilder) cmdSwitchActiveRunEFn(cmd *cobra.Command, args []stri
 	if err != nil {
 		return err
 	}
-	b.name = args[0]
-	p0, ok := pp[b.name]
-	if !ok {
-		return &influxdb.Error{
-			Code: influxdb.ENotFound,
-			Msg:  fmt.Sprintf("name %q is not found", b.name),
+	var p0 config.Config
+	if args[0] == "-" {
+		b.name, p0, err = b.svc.ParsePreviousActiveConfig()
+		if err != nil {
+			return err
+		}
+	} else {
+		b.name = args[0]
```

Comment:
you don't need to assign the name to the bulider here, probably better to not do that actually 🤔. No need to add a side effect here imo 🤷‍♂